### DISCUSSION TITLE
render() return TensorXf values outside the range [0.0,1.0]

### DISCUSSION THREAD

**Discussion by nebnes** [mitsuba-renderer/mitsuba-blender#97] [Q&A] [answered]
After rendering, I get a TensorXf with values outside the range [0.0,1.0]
I get several values above 1.0 around 94.00
It seems out of range values match overly bright pixels. Is it correct to clip the tensor 's high values back to 1.0 ?
`np.clip(img,0.0,1.0)`

Is it correct ? 

Thanks in advance

**Comment by bathal1** [comment, marked as answer]
The output of a rendering in Mitsuba is in a linear color space, so there is no reason for the values to be in the [0, 1] range. 

What you do to get it to this range entirely depends on what you intend to do with the image after rendering. For example, in order to write it in a common format like JPEG or PNG, one would apply tonemapping to convert it to the correct color space.

Why do you need values to be in that range ?

**Reply by nebnes** [reply]
@bathal1 
I didn't particularly expect to be in this range. It's just that I had some big gaps in a rendering. I understand now, thank you.

**Reply by bathal1** [reply]
If you want to know more about how images are handled in Mitsuba, I recommend you take a look at the [image manipulation](https://mitsuba.readthedocs.io/en/latest/src/how_to_guides/image_io_and_manipulation.html) tutorial. The "Bitmap conversion" section might be of particular interest to you.

**Reply by nebnes** [reply]
I'll take a look
 thanks @bathal1
